tests/io/test_pyarrow.py

```diff
@@ -1798,3 +1799,35 @@ def test_identity_partition_on_multi_columns() -> None:
             ("n_legs", "ascending"),
             ("animal", "ascending"),
         ]) == arrow_table.sort_by([("born_year", "ascending"), ("n_legs", "ascending"), ("animal", "ascending")])
+
+
+def test_to_requested_schema_timestamps(
```

Comment:
I noticed that there weren't any tests for `to_requested_schema` although it is a public API, so I added this in.

Do we want to keep `to_requested_schema` as a public API?

Reply:
+1 on making `to_requested_schema` private. IMO this seem to be specific to our internal workflow so making it private can give us more flexibility in future development.

Actually, the signature of this API has been updated in https://github.com/apache/iceberg-python/pull/786: `pa.Table` to `pa.RecordBatch`, which is a breaking change. I think it may be a good time to send a deprecation notice on the old one and making the new one private. WDYT?

Reply:
I've made the function private on this PR, and have another one up to port back to public API function with the previous signature with a deprecation notice here: https://github.com/apache/iceberg-python/pull/918